core/src/main/java/hudson/model/Computer.java

```diff
@@ -606,6 +602,11 @@ public Node getNode() {
         return j.getNode(nodeName);
     }
 
+    @NonNull
+    private Node getNodeOrDie() {
+        return nodeName == null ? Jenkins.get() : Jenkins.get().getNode(nodeName);
```

Comment:
https://github.com/jenkinsci/jenkins/blob/962a286bc67f2f3e65fe2842a1d7b12463d8cf89/core/src/main/java/jenkins/model/Jenkins.java#L2228 is `@CheckForNull` so this method should be too, and callers should in fact check for null.

IIRC it is legitimately possible for there to be a `SlaveComputer` with no corresponding `Slave`, at least in the case of permanent agents: the agent is still running an executor when the node is deleted. (Conversely, you can have a `Slave` with no `SlaveComputer`, if it has zero executors.) It is definitely a corner case and I do not think we need to expose any “temporarily offline” status in that case (the `SlaveComputer` will just be removed as soon as it is done with its current task), but it needs to handled and not throw an exception.